OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2520,3 +2528,72 @@ void testUniversalJointAccessors()
            __FILE__, __LINE__, "Coordinate accessor failed");
     ASSERT_THROW(OpenSim::InvalidCall, myUniversalJoint.updCoordinate());
 }
+
+void testMotionTypesForCustomJointCoordinates()
+{
+    Model osimModel;
+
+    //OpenSim bodies
+    const Ground& ground = osimModel.getGround();
+    //OpenSim thigh
+    auto osim_thigh = new OpenSim::Body ("thigh", femurMass.getMass(),
+        femurMass.getMassCenter(), femurMass.getInertia());
+
+    // Define hip coordinates and axes for custom joint
+    SpatialTransform hipTransform;
+    OpenSim::Array<std::string> coordNames;
+    coordNames.append("hip_qx");
+    coordNames.append("hip_qy");
+    hipTransform[2].setCoordinateNames(coordNames);
+    hipTransform[2].setFunction(new MultidimensionalFunction());
+    hipTransform[3].setCoordinateNames(OpenSim::Array<std::string>(coordNames[0], 1, 1));
+    hipTransform[3].setFunction(new LinearFunction());
```

Comment:
Since a slope of 1 is handled as a special case, perhaps include a case here (regression test) where a `LinearFunction` has a slope of 1 but a nonzero intercept (see comment above).